src/org/usfirst/frc4904/robot/humaninterface/drivers/DriveStraightSpeedModifier.java

```diff
@@ -0,0 +1,45 @@
+package org.usfirst.frc4904.robot.humaninterface.drivers;
+
+
+import org.usfirst.frc4904.standard.LogKitten;
+import org.usfirst.frc4904.standard.Util;
+import org.usfirst.frc4904.standard.custom.ChassisController;
+import org.usfirst.frc4904.standard.custom.motioncontrollers.MotionController;
+import org.usfirst.frc4904.standard.custom.sensors.InvalidSensorException;
+import org.usfirst.frc4904.standard.subsystems.motor.speedmodifiers.SpeedModifier;
+
+public class DriveStraightSpeedModifier implements SpeedModifier {
+	protected final ChassisController controller;
+	protected final MotionController motionController;
+
+	public DriveStraightSpeedModifier(ChassisController controller, MotionController motionController) {
+		this.controller = controller;
+		this.motionController = motionController;
+	}
+
+	@Override
+	public double modify(double speed) {
+		// If we're driving straight, use drive straight PID
```

Comment:
The problem with only running drive straight PID if we are going straight, is that if we are slightly turning towards one side by default (which is what drive straight PID is for), then when we begin to try to turn in the opposite direction it will stop correcting, making it so that it will be still driving straight within a certain range. I feel like if we're going to do drive straight PID at all, we should have it running all of the time.